### ISSUE TITLE
Shadertoy example with frame buffer as texture

### ISSUE THREAD

**Issue by bravokiloecho** [greggman/dekapng#3]
I'm trying to expand on your shadertoy example and make a version where a frame buffer acts as a texture to the main canvas (like you can do with Shadertoy).

I managed to recreate your shadertoy + dekapng example with a correct PNG export, but when I'm trying to run it with the frame buffer texture, the hi-res version isn't a match. I've tried several ways of resizing and offsetting the frame buffer during the render but my brain has hit a wall!

Here is the `draw` call which is used in creating the tiles for dekapng: https://github.com/bravokiloecho/dekapng-shadertoy/blob/2f23ec635c6a1df2e6059321e3fc69827ff15b4d/src/index.js#L59

And the dekapng logic is all in here: https://github.com/bravokiloecho/dekapng-shadertoy/blob/main/src/exportImage.js

The two frag files are: [drawing onto frame buffer](https://github.com/bravokiloecho/dekapng-shadertoy/blob/main/src/frag0.glsl) and [rendering out the frame buffer texture](https://github.com/bravokiloecho/dekapng-shadertoy/blob/main/src/frag1.glsl)

[You can see a live version here](https://dekapng-shadertoy.vercel.app). Press [p] to run the export PNG function

Lastly thank you so much for all your work on this library! It's a really great resource.

**Comment by greggman** [comment]
I didn't look too deeply into your code but making an shadertoy shader that uses a texture generated but another shader isn't going to be as simple to make work as one that doesn't (a single pass shadertoy)

The issue is the 2nd pass could access any part of the 1st pass, not just some small portion of it. 

Imagine the first pass generates

    1122
    3344

And the second pass generates

    AABB
    CCDD

You split the second pass into

    AA  BB
  
    CC  DD

But when generating each of those 4 parts there's nothing in the 2nd pass shader that says only use the corresponding part of the 1st pass. It can access any part of the 1st pass so the entirety of the 1st pass needs to be available, not just some sub portion of it.

In other words, let's say the 2nd pass was

```
void mainImage(out vec4 fragColor,in vec2 fragCoord)
{
  vec2 uv = fragCoord.xy  / iResolution.xy;
  
  vec3 col = texture2D(iChannel0, vec2(1) - uv).rgb;
  
  fragColor=vec4(col,1.);
}
```

Now, when generating section AA of the 2nd pass, it's looking at section 44 of the first pass. But in your code, only section 11 of the first pass is being passed to the 2nd pass

I think you need to generate the first pass at some fixed resolution. Generate it just once, outside the loop at say 2048x2048, then pass the entire thing as iChannel0 to the chunks of the 2nd pass.
